internal/templates/danglingdeployment/template.go

```diff
@@ -0,0 +1,59 @@
+package danglingdeployment
+
+import (
+	"fmt"
+
+	"golang.stackrox.io/kube-linter/internal/extract"
+
+	"golang.stackrox.io/kube-linter/internal/check"
+	"golang.stackrox.io/kube-linter/internal/diagnostic"
+	"golang.stackrox.io/kube-linter/internal/lintcontext"
+	"golang.stackrox.io/kube-linter/internal/objectkinds"
+	"golang.stackrox.io/kube-linter/internal/templates"
+	"golang.stackrox.io/kube-linter/internal/templates/danglingdeployment/internal/params"
+	appsV1 "k8s.io/api/apps/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Dangling Deployments",
+		Key:         "dangling-deployment",
+		Description: "Flag deployments which do not have any selector matching a label in pod template",
+		SupportedObjectKinds: check.ObjectKindsDesc{
+			ObjectKinds: []string{objectkinds.DeploymentLike},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(_ params.Params) (check.Func, error) {
+			return func(_ *lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
+				deployment, ok := object.K8sObject.(*appsV1.Deployment)
+				if !ok {
+					return nil
+				}
```

Comment:
So, this check is applicable not just to deployments, but also to Daemonsets, Jobs etc. Can you add a new extractor for this, and implement it similar to the `PodTemplateSpec` extractor? ie, something like:
```go
func Selector(obj k8sutil.Object) (*metaV1.LabelSelector, bool) {
	switch obj := obj.(type) {
	case *batchV1Beta1.CronJob:
		return obj.Spec.JobTemplate.Spec.Selector, true
	default:
		objValue := reflect.Indirect(reflect.ValueOf(obj))
		spec := objValue.FieldByName("Spec")
		if !spec.IsValid() {
			return coreV1.PodTemplateSpec{}, false
		}
		selector := spec.FieldByName("Selector")
		if !selector.IsValid() {
			return nil, false
		}
		labelSelector, ok := template.Interface().(*metaV1.LabelSelector)
		if ok {
			return labelSelector, true
		}
	}
        return nil, false
}
```

and then you can use this selector below.
(Caveat: just typed out the code above, you may need to tweak it to get it to compile)

Reply:
Done